test/functionalTest/cases/2507_csub_metadata_field/metadata_filter.test

```diff
@@ -0,0 +1,792 @@
+echo "=========================================================="
+payload='{
+  "type": "T",
+  "id": "E",
+  "A": {
+    "type": "Number",
+    "value": 1,
+    "metadata": {
+      "M1": { "value": "10", "type": "Number" },
+      "M2": { "value": "20", "type": "Number" },
+      "M3": { "value": "30", "type": "Number" }
+    }
+  },
+  "B": {
+    "type": "Number",
+    "value": 2,
+    "metadata": {
+      "M2": { "value": "200", "type": "Number" }
+    }
+  },
+  "C": {
+    "type": "Number",
+    "value": 3,
+    "metadata": {
+      "M3": { "value": "300", "type": "Number" }
+    }
+  },
+  "D": {
+    "type": "Number",
+    "value": 4
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo "06. Dump and reset: see 4 notification (M1+M2+M3, M1+M2+M3+actionType, M1+M2, actionType+M3)"
+echo "============================================================================================"
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+echo "07. Update entity attribute"
+echo "==========================="
+payload='{
+  "D": {
+    "value": "x",
+    "type": "Text"
+  }
+}'
+orionCurl --url /v2/entities/E/attrs -X PATCH --payload "$payload"
+echo
+echo
+
+
+echo "08. Dump and reset: see again 4 notification (M1+M2+M3, M1+M2+M3+actionType, M1+M2, actionType+M3), actionType only for D"
+echo "========================================================================================================================="
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+--REGEXPECT--
+01. Create sub1 with metadata: *
+================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Create sub2 with metadata: M1, M2
+=====================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. Create sub3 with metadata: actionType, M3
+=============================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+04. Create sub4 with metadata: actionType, *
+=============================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+05. Create entity with attributes with metadata M1, M2, M3
+==========================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+06. Dump and reset: see 4 notification (M1+M2+M3, M1+M2+M3+actionType, M1+M2, actionType+M3)
+============================================================================================
+#SORT_START
```

Comment:
Not fully sure about the order... I think I used a reference execution (copy-paste :) to fill that part, then use SORT when I realize after some executions that the order actually was slightly different from .test to .test.